.github/workflows/pr-checks.yml

```
@@ -0,0 +1,107 @@
+      codecov: true
+
+  build-binaries:
+    name: Build Binaries
+    needs: [ static-analysis, unit-tests, integration-tests ]
+    uses: ./.github/workflows/_build.yml
+    secrets: inherit
+    if: success()
+
+  container-tests:
+    name: Container Tests
+    if: env.SHOULD_RUN == 'true'
```

actionlint (1.7.4)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)